From 6803cfce89a1ee6a7b6e7c4d1bcd300459d28774 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Wed, 07 Jun 2023 11:31:53 +0300
Subject: [PATCH] [SW-142278] h6: tpm: remove tpm_en efuse bit check

As part of SW-117389, tpm init is moved from erom to ppboot to meet pcie
timing constraints. So ROM can't expect TPM on EROM and tpm_en efuse bit
is not burnt. erom tpm boot descriptors were removed from image pointers
so that arcboot rom skips calling erom tpm always.

The decision was to support TPM feature independent of tpm_en efuse bit
in all FW components. So ppboot runs tpm code unconditionally

Change-Id: Ia2f2118a400dda99284374d6837a6449179fd43e
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/arch/gaudi2/common/tpm_auth.c b/arch/gaudi2/common/tpm_auth.c
index a7eed13..0adfc82 100644
--- a/arch/gaudi2/common/tpm_auth.c
+++ b/arch/gaudi2/common/tpm_auth.c
@@ -15,7 +15,6 @@
 #include "lzo/unaligned.h"
 
 #define TPM_NV_DEV_SN_ADDRESS		0x01c10100
-#define EFUSE_SYSCFG_ENABLE_TPM_MASK	BIT(11)
 
 struct tpm_dev_id {
 	union {
@@ -39,26 +38,9 @@
 	return 0;
 }
 
-static int tpm_is_enabled(void)
-{
-	u32 config_fuse;
-	int ret = preboot_fuse_read(EFUSE_SYSCFG_OFFSET, &config_fuse);
-
-	if (ret)
-		return ret;
-
-	if ((config_fuse & EFUSE_SYSCFG_ENABLE_TPM_MASK) == 0)
-		return -1;
-
-	return 0;
-}
-
 int tpm_init(bool print)
 {
 	int ret;
-
-	if (tpm_is_enabled() != 0)
-		return -EOPNOTSUPP;
 
 	ret = tpm2_init(print);
 	if (ret) {
@@ -78,9 +60,6 @@
 {
 	u8 capabilities[TPM2_CAP_SIZE];
 	int ret;
-
-	if (tpm_is_enabled() != 0)
-		return -EOPNOTSUPP;
 
 	ret = tpm2_startup(TPM2_SU_CLEAR);
 	if (ret) {
@@ -122,9 +101,6 @@
 {
 	int ret;
 
-	if (tpm_is_enabled() != 0)
-		return -EOPNOTSUPP;
-
 	if (hash == NULL)
 		return -1;
 
@@ -144,9 +120,6 @@
 	struct tpm_dev_id efuse_dev_id;
 
 	efuse_dev_id.tpm_data = 0x0;
-
-	if (tpm_is_enabled() != 0)
-		return;
 
 	if (cpu_boot_err_get(CPU_BOOT_ERR_TPM_FAIL))
 		return;
